crates/net/common/src/ban_list.rs

```diff
@@ -18,18 +31,22 @@ impl BanList {
         banned_peers: impl IntoIterator<Item = PeerId>,
         banned_ips: impl IntoIterator<Item = IpAddr>,
     ) -> Self {
-        Self {
-            banned_ips: banned_ips.into_iter().map(|ip| (ip, None)).collect(),
-            banned_peers: banned_peers.into_iter().map(|peer| (peer, None)).collect(),
-        }
+        Self::new_with_timeout(
+            banned_peers.into_iter().map(|peer| (peer, None)).collect(),
+            banned_ips.into_iter().map(|ip| (ip, None)).collect(),
+        )
     }
 
     /// Creates a new ban list that bans the given peers and ips with an optional timeout.
     pub fn new_with_timeout(
         banned_peers: HashMap<PeerId, Option<Instant>>,
         banned_ips: HashMap<IpAddr, Option<Instant>>,
     ) -> Self {
-        Self { banned_ips, banned_peers }
+        // filter out non globally routable IP addresses
+        Self {
+            banned_ips: banned_ips.into_iter().filter(|(ip, _)| is_global(ip)).collect(),
+            banned_peers,
```

Comment:
I think using the input values as is is sufficient if the caller doesn't want non global IPs it can filter them outside

Reply:
ok, that sounds reasonable, will remove the filter and document this behavior